public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -475,6 +475,16 @@ export function buildGridItemForLibPanel(panel: PanelModel) {
   });
 }
 
+const hoverHeaderOffsetBehavior = (grid: DashboardGridItem) => {
+  grid.subscribeToState(() => {
+    grid.forEachChild((child) => {
+      if (child instanceof VizPanel && child.state.hoverHeader) {
+        child.setState({ hoverHeaderOffset: grid.state.y === 0 ? 0 : undefined });
```

Comment:
Can we optimise this and only run `child.setState` if the `y` state of the grid item changed? `subscribeToState` receives new and previous state to perform comparison.